lib/pure/strutils.nim

```diff
@@ -2313,6 +2313,54 @@ proc removeSuffix*(s: var string, suffix: string) {.
     newLen -= len(suffix)
     s.setLen(newLen)
 
+proc removePrefix*(s: var string, chars: set[char] = Newlines) {.
+  rtl, extern: "nsuRemovePrefixCharSet".} =
+  ## Removes the first matching character from the start of a string (in-place)
+  ## given a set of characters. If the set of characters is only equal to
+  ## `Newlines` then it will remove both the newline and return feed.
+  ## .. code-block:: nim
+  ##   var
+  ##     userInput = "\r\n*Hello World!"
+  ##     otherInput = "?!?Hello!?!"
+  ##   userInput.removePrefix
+  ##   userInput == "*Hello World!"
+  ##   userInput.removePrefix({'H', '*'})
+  ##   userInput == "Hello World!"
+  ##   otherInput.removePrefix({'!', '?'})
+  ##   otherInput == "!?Hello!?!"
+  if s.len == 0: return
+  var start = 0
+  if chars == Newlines:
```

Comment:
This is confusing: 

- Newlines should not be special cased.
- This should do:  

while s[start] in chars: start += 1

Reply:
I had followed the existing `removeSuffix` implementation, which removes only the first occurrence. Your proposed change would be a pretty big shift in behavior since instead it would remove all occurrences at the start:

| input | current | your proposal |
| :---: | :---: | :---: |
| `removePrefix("\n\n\n\n\nhello")` | `"\n\n\n\nhello"` | `"hello"` |
| `removePrefix("hello world", {'h', 'e', 'l'})` | `"ello world"` | `"o world"` |

If this is desired then I'd think `removeSuffix` should definitely be updated to match.

Reply:
I suppose ``removeSuffix`` slipped through my review. Procs that take a ``set[char]`` should generally match a regex like ``[<set here>]*``, everything else is confusing. ``removePrefix("   abc", Whitespace)`` should remove leading whitespace, not the first char that is whitespace.

Reply:
@Araq I can agree with that. Should I make your change for `removePrefix` here and submit a new PR for updating `removeSuffix`?

Reply:
yes please.

Reply:
@Araq done. Once this merges I'll open a new PR for `removeSuffix` :+1: